packages/applicant/src/pages/submissions/[submissionId]/equality-and-diversity/[grantBeneficiaryId]/organisation.page.tsx

```diff
@@ -0,0 +1,152 @@
+import callServiceMethod from '../../../../../utils/callServiceMethod';
+import { getJwtFromCookies } from '../../../../../utils/jwt';
+import {
+  errorPageParams,
+  errorPageRedirect,
+} from '../equality-and-diversity-service-errors';
+import { EqualityAndDiversityParams } from '../types';
+import { fetchGrantBeneficiary } from './fetchGrantBeneficiary';
+
+export enum OrganisationRadioOptions {
+  VCSE = 'Voluntary, community and social enterprise',
+  SME = 'Small to medium enterprise',
+  NEITHER = 'Neither',
+}
+
+type RequestBody = {
+  organisation: `${OrganisationRadioOptions}`;
+};
+
+export const getServerSideProps: GetServerSideProps<{}, EqualityAndDiversityParams> = async ({
+  params,
+  resolvedUrl,
+  req,
+  res,
+  query,
+}) => {
+  const { submissionId, grantBeneficiaryId } = params as Record<string, string>;
+  const { returnToSummaryPage } = query as Record<string, string>;
+  let defaultChecked = null;
+
+  let grantBeneficiary: GrantBeneficiary;
+  try {
+    grantBeneficiary = await fetchGrantBeneficiary(submissionId, req);
+  } catch (_) {
+    return errorPageRedirect(submissionId);
+  }
+
+  if (grantBeneficiary.organisationGroup1) defaultChecked = OrganisationRadioOptions.VCSE;
+  else if (grantBeneficiary.organisationGroup2) defaultChecked = OrganisationRadioOptions.SME;
+  else if (grantBeneficiary.organisationGroup3) defaultChecked = OrganisationRadioOptions.NEITHER;
+
+  const response = await callServiceMethod(
+    req,
+    res,
+    async (body: RequestBody) => {
+      if (body.organisation) {
+        await postGrantBeneficiaryResponse(
+          {
+            submissionId: submissionId,
+            hasProvidedAdditionalAnswers: true,
+            organisationGroup1:
+              body.organisation === OrganisationRadioOptions.VCSE,
+            organisationGroup2:
+              body.organisation === OrganisationRadioOptions.SME,
+            organisationGroup3: body.organisation === OrganisationRadioOptions.NEITHER,
+          },
+          getJwtFromCookies(req),
+          grantBeneficiaryId
+        );
+      }
+    },
+    returnToSummaryPage
+      ? `/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/summary`
+      : `/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/sex`,
+    errorPageParams(submissionId)
+  );
+
+  if ('redirect' in response) {
+    return response;
+  }
+
+  const { publicRuntimeConfig } = getConfig();
+
+  return {
+    props: {
+      formAction: `${publicRuntimeConfig.subPath}${resolvedUrl}`,
+      skipURL: `${
+        publicRuntimeConfig.subPath
+      }/submissions/${submissionId}/equality-and-diversity/${grantBeneficiaryId}/${
+        returnToSummaryPage ? 'summary' : 'sex'
+      }`,
+      defaultChecked: defaultChecked,
+      csrfToken: (req as any).csrfToken?.() || '',
+    },
+  };
+};
+
+const OrganisationPage = ({
+  formAction,
+  defaultChecked,
+  skipURL,
+  csrfToken,
+}: OrganisationPageProps) => {
```

Comment:
Appreciate its been done like this in all the other pages, but we can maintain type safety between getServerSideProps and pages using the type `InferProps<typeof getServerSideProps>`.  Sometimes have to modify the type of getServerSideProps to get this to infer things correctly tho

Reply:
That's less typing but it does remove VSCode's ability to tell you what each individual prop should be on hover. I wonder if there's a way to do both :thinking:

Reply:
So it behaves pretty reasonably if you set getServerSideProps props to `GetServerSidePropsContext` instead of typing the entire function. Obvs with the disadvantage that the return types can be anything :( yet to find a perfect solution

Reply:
I don't think it's possible to correctly specify the prop types without defining them somewhere - even if you were to infer them from `getServerSideProps` they would have had to have been passed into the function type definition to be able to get them out the other end.

Have amended the function type to `GetServerSideProps<OrganisationPageProps, EqualityAndDiversityParams>` for a bit more type safety though.